src/CosmosDB/CosmosDB/SQL/UpdateAzCosmosDbClientEncryptionKey.cs

```diff
@@ -0,0 +1,215 @@
+{
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDbClientEncryptionKey", DefaultParameterSetName = NameParameterSet, SupportsShouldProcess = true), OutputType(typeof(PSSqlClientEncryptionKeyGetResults), typeof(ResourceNotFoundException))]
+    public class UpdateAzCosmosDbClientEncryptionKey : AzureCosmosDBCmdletBase
+    {
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSet, HelpMessage = Constants.ResourceGroupNameHelpMessage)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSet, HelpMessage = Constants.AccountNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string AccountName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSet, HelpMessage = Constants.DatabaseNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string DatabaseName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.ClientEncryptionKeyName)]
+        [ValidateNotNullOrEmpty]
+        public string ClientEncryptionKeyName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.EncryptionAlgorithmName)]
+        [ValidateNotNullOrEmpty]
+        public string EncryptionAlgorithmName { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipeline = true, HelpMessage = Constants.WrappedDataEncryptionKey)]
+        [ValidateNotNullOrEmpty]
+        public byte[] WrappedDataEncryptionKey { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipeline = true, HelpMessage = Constants.KeyWrapMetaData)]
+        [ValidateNotNullOrEmpty]
+        public PSSqlKeyWrapMetadata KeyWrapMetadata { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.IsAzureKeyVaultKeyStoreProvider)]
+        [ValidateNotNullOrEmpty]
+        public bool IsAzureKeyVaultKeyStoreProvider { get; set; } = true;
+
+        [Parameter(Mandatory = false, ValueFromPipeline = true, HelpMessage = Constants.IsAzureKeyVaultKeyStoreProvider)]
+        [ValidateNotNullOrEmpty]
+        public EncryptionKeyStoreProvider EncryptionKeyStoreProvider { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParentObjectParameterSet, HelpMessage = Constants.SqlDatabaseObjectHelpMessage)]
+        [ValidateNotNull]
+        public PSSqlDatabaseGetResults ParentObject { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ObjectParameterSet, HelpMessage = Constants.ClientEncryptionKeyObjectHelpMessage)]
+        [ValidateNotNull]
+        public PSSqlClientEncryptionKeyGetResults InputObject { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName.Equals(ParentObjectParameterSet, StringComparison.Ordinal))
```

Comment:
What does this mean? Update would be on single key, correct?

Reply:
Pass the parent database object.